### PR TITLE
telemetry-gateway: copy range variable

### PR DESCRIPTION
**Pull request by bobheadxi**
Not only did I fall for the oldest mistake in the book, but I [couldn't write a good test to reproduce it](https://github.com/sourcegraph/sourcegraph/pull/57236) and went on a goose hunt other possible sources of this bug thinking my originally bad test had ruled it out 😮‍💨 

## Test plan

Updated test.

### CODE REVIEW

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* main
  * **PR #57236** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/57236" target="_blank">Graphite</a> 
    * **PR #57237** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/57237" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/57237?utm_source=stack-comment).

File: cmd/telemetry-gateway/internal/events/events.go [hunk 1]
```diff
@@ -37,11 +37,14 @@ type PublishEventResult struct {
 	PublishError error
 }
 
+// Publish emits all events concurrently, up to 100 at a time for each call.
 func (p *Publisher) Publish(ctx context.Context, events []*telemetrygatewayv1.Event) []PublishEventResult {
 	wg := pool.NewWithResults[PublishEventResult]().
 		WithMaxGoroutines(100) // limit each batch to some degree
 
 	for _, event := range events {
+		event := event // capture range variable :(
```
Comment (bobheadxi) [inline cmd/telemetry-gateway/internal/events/events.go:46]: This was the bug

**Review by nathan-downs — APPROVED** [review]

File: cmd/telemetry-gateway/internal/events/events.go (hunk 1 above)
Comment (rafax) [reply cmd/telemetry-gateway/internal/events/events.go:46]: https://github.com/golang/go/wiki/LoopvarExperiment Go will hopefully help you in the future ;)

**Review by unknwon — APPROVED** [review]

**Comment by unknwon** [comment]
So glad we caught this!!

**Pull request merged**